mockgcp/mockprivateca/capool.go

```diff
@@ -53,16 +62,44 @@ func (s *PrivateCAV1) CreateCaPool(ctx context.Context, req *pb.CreateCaPoolRequ
 		return nil, err
 	}
 
+	now := time.Now()
+
 	fqn := name.String()
 
 	obj := proto.Clone(req.CaPool).(*pb.CaPool)
 	obj.Name = fqn
 
+	// service seems to remove "zero" values
+	baseKeyUsage := obj.GetIssuancePolicy().GetBaselineValues().GetKeyUsage().GetBaseKeyUsage()
+	if baseKeyUsage != nil {
+		if proto.Equal(baseKeyUsage, &pb.KeyUsage_KeyUsageOptions{}) {
+			obj.IssuancePolicy.BaselineValues.KeyUsage.BaseKeyUsage = nil
+		}
+	}
+
+	extendedKeyUsage := obj.GetIssuancePolicy().GetBaselineValues().GetKeyUsage().GetExtendedKeyUsage()
+	if extendedKeyUsage != nil {
+		if proto.Equal(extendedKeyUsage, &pb.KeyUsage_ExtendedKeyUsageOptions{}) {
+			obj.IssuancePolicy.BaselineValues.KeyUsage.ExtendedKeyUsage = nil
+		}
+	}
+
 	if err := s.storage.Create(ctx, fqn, obj); err != nil {
 		return nil, err
 	}
 
-	return s.operations.NewLRO(ctx)
+	opMetadata := &pb.OperationMetadata{
+		ApiVersion:            "v1",
+		CreateTime:            timestamppb.New(now),
+		Verb:                  "create",
+		RequestedCancellation: false,
+		Target:                fqn,
+	}
+	opPrefix := fmt.Sprintf("projects/%s/locations/%s", name.Project.ID, name.Location)
+	return s.operations.StartLRO(ctx, opPrefix, opMetadata, func() (proto.Message, error) {
+		opMetadata.EndTime = timestamppb.Now()
```

Comment:
This doesn't really matter, but I like that you putting it in the callback than the initialization. (I shall follow that as well).